bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuntimeCondition.java

```diff
@@ -26,7 +27,8 @@
     private Set<Connection> connections;
 
     public RuntimeCondition(Condition condition) {
-        super(condition.getId(), condition.getTypeUID(), condition.getConfiguration(), condition.getInputs());
+        super(condition.getId(), condition.getTypeUID(),
+                new Configuration(condition.getConfiguration().getProperties()), condition.getInputs());
```

Comment:
This looks pretty ugly to have in all the constructors.
Two questions:
1. Why do we need a copy of the passed configuration at all?
2. If we need it, why isn't this simply done once in `Module.setConfiguration()`?

Reply:
Changing setConfiguration  visibility to private is backward incompatible change of java api. Also I think in that way the user has opportunity to change the configuration of the module several times  without creating a new module instance, but this can be happen only before adding the rule into rule engine. After that the user can't change module's configuration properties until the rule's update operation is called.

Reply:
Not sure why this then speaks against doing the copy within `Module.setConfiguration()`...?

Reply:
@adimova Could you comment on this?

Reply:
I agree to copy the passed configuration and this to be done once in `Module.setConfiguration()`.
This is already done in #4468.